blogs/2022/04/04/increase-productivity-with-containers.md

```diff
@@ -0,0 +1,88 @@
+
+## Moving to the cloud
+
+Let’s be real: everything is moving to the cloud, so why should your development environment be any different? Well, for one, the cloud is vulnerable to attacks. For two, it seems like we’re constantly hearing about major outages. For three, it can be pricey. For four… wait, isn’t the cloud supposed to be a good thing?
+
+With all these reasons to be weary of the cloud, it can be uncomfortable to think about something as important as our development environment being hosted there.  So let’s stay comfortable! Let’s continue relying on our 6-year-old laptop that makes really weird whirring noises if you try to open an email too quickly, let alone try to build a project. And we’ll just comfortably wait for the inevitability of when the laptop crashes, and then once we get our new laptop, we have to rebuild our dev environment and try to remember exactly how we had it configured.
+
+Turns out staying in our comfort zone doesn’t sound so peachy either.
+
+While there are inherent risks with the cloud, it also comes with major benefits: scalability, faster performance, and easier maintenance, just to name a few. Hosting in the cloud gives you access to another machine’s resources that can then be spun up quickly and available anywhere. Combine that with the benefits of container-based development, and developers can be up and coding in the blink of an eye.
+
+## Containers in the cloud
+
+Running containers in the cloud is not a new concept; in fact, [one study](https://www.datadoghq.com/container-report/?utm_source=SocialMedia&utm_medium=Twitter&utm_campaign=OrganicPosting-containerreport) shows that at least half of all containers are running in the cloud. The basic infrastructure involves deploying a container to a cloud-hosted VM. We can deploy our dev container to the cloud  to provide a cloud-hosted development environment.
+
+VS Code’s entry into this space comes from powering [GitHub Codespaces](https://github.com/features/codespaces). Within minutes, you can create and configure a dev container hosted in the cloud that is ready whenever you need it. You can then connect via VS Code (in either the browser or desktop) to a development environment that is managed completely for you, no longer relying on your laptop’s resources to handle the demand.
+
+![A diagram showing the infrastructure and workflow of Codespaces](codespaces-diagram.png)
+
+The same dev containers used in the Remote - Containers extension can be used in Codespaces, providing a seamless transition to the cloud.
+
+But remember all of those scary possibilities of the cloud? Well, Codespaces alleviates those issues by leveraging Github's cloud features. Codespaces run on compute options hosted on Github.com, and the feature is currently available for those using [GitHub Team or GitHub Enterprise Cloud](https://docs.github.com/en/get-started/learning-about-github/githubs-products).
```

Comment:
@bamurtaugh here's where i begin addressing the cloud concerns!

Reply:
Love this! I wonder if teasing this earlier would help (i.e mentioning, hey but there are steps we can take to mitigate these issues, which you'll learn more about shortly)